File: php/sendTelegram.php
<?php
header('Content-Type: application/json');

if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
    echo json_encode(['success' => false, 'message' => 'Invalid request']);
    exit;
}

// Honeypot: if filled, silently succeed
if (!empty($_POST['website'])) {
    echo json_encode(['success' => true, 'message' => 'Message sent!']);
    exit;
}

$name    = isset($_POST['name'])    ? trim(strip_tags($_POST['name']))    : '';
$message = isset($_POST['message']) ? trim(strip_tags($_POST['message'])) : '';

if (empty($name) || empty($message)) {
    echo json_encode(['success' => false, 'message' => 'Name and message are required']);
    exit;
}

if (strlen($message) > 2000) {
    echo json_encode(['success' => false, 'message' => 'Message too long']);
    exit;
}

// Load credentials: prefer /home/private/.env, fall back to repo .env
$env_paths = ['/home/private/.env', dirname(dirname(__FILE__)) . '/.env'];
$token = null;
$chat_id = null;

foreach ($env_paths as $path) {
    if (file_exists($path)) {
        $env = parse_ini_file($path);
        if (!empty($env['TELEGRAM_BOT_TOKEN']) && !empty($env['TELEGRAM_CHAT_ID'])) {
            $token   = $env['TELEGRAM_BOT_TOKEN'];
            $chat_id = $env['TELEGRAM_CHAT_ID'];
            break;
        }
    }
}

if (!$token || !$chat_id) {
    echo json_encode(['success' => false, 'message' => 'Server configuration missing']);
    exit;
}

$text = "Message from morganrivers.com\nName: " . $name . "\n\n" . $message;

$ch = curl_init("https://api.telegram.org/bot" . $token . "/sendMessage");
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, ['chat_id' => $chat_id, 'text' => $text]);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_TIMEOUT, 10);
$result = curl_exec($ch);
$err    = curl_error($ch);
curl_close($ch);

if ($err || $result === false) {
    echo json_encode(['success' => false, 'message' => 'Failed to send message']);
} else {
    $resp = json_decode($result, true);
    if ($resp && $resp['ok']) {
        echo json_encode(['success' => true, 'message' => 'Message sent!']);
    } else {
        echo json_encode(['success' => false, 'message' => 'Telegram API error']);
    }
}
?>

File: cgi-bin/sendTelegram.py
#!/usr/bin/env python3
import cgi
import json
import os
import urllib.request
import urllib.parse

print("Content-Type: application/json")
print()

def load_env(path):
    env = {}
    try:
        with open(path) as f:
            for line in f:
                line = line.strip()
                if '=' in line and not line.startswith('#'):
                    k, v = line.split('=', 1)
                    env[k.strip()] = v.strip()
    except:
        pass
    return env

env = load_env('/home/protected/.env')
if not env:
    env = load_env(os.path.join(os.path.dirname(os.path.dirname(__file__)), '.env'))

token   = env.get('TELEGRAM_BOT_TOKEN')
chat_id = env.get('TELEGRAM_CHAT_ID')

if not token or not chat_id:
    print(json.dumps({'success': False, 'message': 'Server configuration missing'}))
    exit()

form     = cgi.FieldStorage()
name     = form.getvalue('name', '').strip()
message  = form.getvalue('message', '').strip()
honeypot = form.getvalue('website', '')

if honeypot:
    print(json.dumps({'success': True, 'message': 'Message sent!'}))
    exit()

if not name or not message:
    print(json.dumps({'success': False, 'message': 'Name and message are required'}))
    exit()

if len(message) > 2000:
    print(json.dumps({'success': False, 'message': 'Message too long'}))
    exit()

text = f"Message from morganrivers.com\nName: {name}\n\n{message}"
data = urllib.parse.urlencode({'chat_id': chat_id, 'text': text}).encode()

try:
    req = urllib.request.Request(f"https://api.telegram.org/bot{token}/sendMessage", data=data)
    with urllib.request.urlopen(req, timeout=10) as resp:
        result = json.loads(resp.read())
    if result.get('ok'):
        print(json.dumps({'success': True, 'message': 'Message sent!'}))
    else:
        print(json.dumps({'success': False, 'message': 'Telegram API error'}))
except Exception:
    print(json.dumps({'success': False, 'message': 'Failed to send message'}))
